components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/UnsupportedAddonsAdapter.kt

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.addons.ui
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageButton
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import mozilla.components.feature.addons.Addon
+import mozilla.components.feature.addons.AddonManager
+import mozilla.components.feature.addons.R
+
+/**
+ * An adapter for displaying unsupported add-on items.
+ *
+ * @property addonManager Manager of installed and recommended [Addon]s and manages their states.
+ * @property unsupportedAddonsAdapterDelegate Delegate that will provides method for handling the
+ * callbacks of uninstalling an add-on.
```

Comment:
nit: The delegate could in the future handle more than just uninstalling. Maybe "...that will provides callbacks for handling any interactions with the unsupported add-ons to the app to handle". What do you think? :)

Reply:
I agree @Amejia481. Making the delegate more generic is not a bad idea :)

Reply:
Fixed.